### DISCUSSION TITLE
Can I receive raw pixel data as compressed image data?

### DISCUSSION THREAD

**Discussion by kimcoldwater** [FreeRDP/FreeRDP#8209] [Q&A] [answered]
Hello.

Can I get the raw pixel data (`gdi->primary_buffer`) of `end_paint` as compressed data such as jpg? ( file or buffer )

Since the image on the rdp screen is required, I tried to convert it to an image using the C library, such as `stb_image`.

In `end_paint`.

However, this had a significant impact on performance and reliability and could not be used.

I've got a couple of other options in mind, but it's going to take a lot of time.

Is there a way to make jpg in freerdp?

I saw that freerdp contains ffmpeg, and I don't know the purpose of the jpeg codec,

 but I asked you because I was wondering if there was a way.

Thank you for reading my question.

**Comment by akallabeth** [comment, marked as answer]
No.
FreeRDP (client) decodes the RDP stream into a bitmap (of same size as all monitors/windows of local client)
You can, of course, compress to any format you like later on, but as you already realized this is not without significant performance penalty.
